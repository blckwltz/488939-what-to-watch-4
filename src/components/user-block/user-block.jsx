import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {getAuthorization} from '../../store/user/selectors.js';
import {AppRoute} from '../../routing/route.js';

const UserBlock = (props) => {
  const {isAuthorized} = props;

  return <div className="user-block">
    {isAuthorized ? <Link to={AppRoute.FAVORITE}><div className="user-block__avatar">
      <img src="/img/avatar.jpg" alt="User avatar" width="63" height="63"/>
    </div></Link> : <Link to={AppRoute.LOGIN} className="user-block__link">Sign in</Link>}
  </div>;
};

UserBlock.propTypes = {
  isAuthorized: PropTypes.bool.isRequired,
};

const mapStateToProps = (state) => ({
  isAuthorized: getAuthorization(state),
});

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);
