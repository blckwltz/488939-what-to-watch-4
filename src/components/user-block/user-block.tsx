import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom';
import {getAuthorization} from '../../store/user/selectors';
import {AppRoute} from '../../routing/route';

interface Props {
  isAuthorized: boolean;
}

const UserBlock = (props: Props) => {
  const {isAuthorized} = props;

  return <div className="user-block">
    {isAuthorized ? <Link to={AppRoute.FAVORITE}><div className="user-block__avatar">
      <img src="/img/avatar.jpg" alt="User avatar" width="63" height="63"/>
    </div></Link> : <Link to={AppRoute.LOGIN} className="user-block__link">Sign in</Link>}
  </div>;
};

const mapStateToProps = (state) => ({
  isAuthorized: getAuthorization(state),
});

export {UserBlock};
export default connect(mapStateToProps)(UserBlock);
