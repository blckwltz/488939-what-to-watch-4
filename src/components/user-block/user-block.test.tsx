import React from 'react';
import renderer from 'react-test-renderer';
import {MemoryRouter} from 'react-router-dom';
import {UserBlock} from './user-block';

it(`Should render UserBlock component correctly`, () => {
  const tree = renderer
    .create(
        <MemoryRouter>
          <UserBlock
            isAuthorized={true}
          />
        </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
