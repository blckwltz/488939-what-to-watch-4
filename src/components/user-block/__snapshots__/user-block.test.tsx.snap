// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render UserBlock component correctly 1`] = `
<div
  className="user-block"
>
  <a
    href="/mylist"
    onClick={[Function]}
  >
    <div
      className="user-block__avatar"
    >
      <img
        alt="User avatar"
        height="63"
        src="/img/avatar.jpg"
        width="63"
      />
    </div>
  </a>
</div>
`;
